### PR TITLE
Adds an additional "to" parameter to sign-in page

### LINKED ISSUE #8587: Login page should support a "to" parameter to customize post-login redirect
In discussion with Cerner team for their use of the static login page, they would like to pass a query parameter `to` to the login page, in addition to the `application` parameter. The `to` parameter would indicate where the user should be redirected to after signing in.

Allowing an entirely free-form redirect is a bad idea because of risks described here: https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html

So, the base domain of the redirect should continue to be based on the configured value of the `application` parameter, with the `to` parameter only specifying the path part of the redirect.

Any other mitigations suggested in that OWASP guideline (such as stripping newlines) should be included.

Additionally, the Cerner team is encoding a second `to` parameter in their URL (for an additional redirect after our redirect), so we need to make sure whatever double-URL encoding is necessary to support that works correctly.

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description
Adds support for an additional parameter `to` that can be specified along with `application` in the signin process. The to parameter is intended as a relative path that gets tacked on to the redirect URL derived from application, to construct a complete redirect URL.

Fixes https://github.com/department-of-veterans-affairs/va.gov-team/issues/8587
in conjunction with https://github.com/department-of-veterans-affairs/vets-api/pull/4304

## Testing done
So far tested locally with sample redirect parameters described in the originating issue. See the corresponding vets-api PR for description of redirect mitigation that occurs there.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/login/containers/SignInApp.jsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ class SignInPage extends React.Component {
     const { query } = this.props.location;
     const loggedOut = query.auth === 'logged_out';
     const application = query.application;
+    const redirect = query.to;
```
Comment (ncksllvn) [inline src/applications/login/containers/SignInApp.jsx:64]: Does this allow relative paths only? Like, if someone types `?to=https://malicious-site.com`, that wouldn't redirect?

Comment (patrickvinograd) [reply src/applications/login/containers/SignInApp.jsx:64]: Yes, but that's enforced in the corresponding vets-api PR (link in description) where we assemble the full redirect path from a base URL derived from the `application` parameter plus the `to` parameter.

**Review by ncksllvn — APPROVED** [review]
LGTM!

File: src/platform/user/authentication/utilities.js [hunk 2]
```diff
@@ -14,25 +14,30 @@ export const ssoKeepAliveEndpoint = () => {
...
+) {
   const SESSIONS_URI =
     version === 'v1'
       ? `${environment.API_URL}/v1/sessions`
       : `${environment.API_URL}/sessions`;
 
   return `${SESSIONS_URI}/${type}/new${
     application ? `?application=${application}` : ''
-  }`;
+  }${application && to ? `&to=${to}` : ''}`;
```
Comment (hinzed1127) [inline src/platform/user/authentication/utilities.js:30]: this is starting to look a little ugly. I'm gonna refactor this into using `URLSearchParams` so it's more readable

Comment (hinzed1127) [reply src/platform/user/authentication/utilities.js:30]: also, for the purposes of unit testing: is it the proper intent that we disregard `to` params unless they're paired with an `application` param, too?

Comment (patrickvinograd) [reply src/platform/user/authentication/utilities.js]: yeah I was not proud of that particular line of code

Comment (ericbuckley) [reply src/platform/user/authentication/utilities.js]: merge in master, I updated this logic in https://github.com/department-of-veterans-affairs/vets-website/pull/12623

**Review by ericbuckley — APPROVED** [review]
LGTM

**Review by U-DON — APPROVED** [review]

**Pull request merged**